the-new-octal/src/app/service/product.service.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Product } from '../model/product';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
+
+  apiUrl: string = `${environment.apiUrl}product`;
+  endPoint: string = 'product';
+
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  getAll(): Observable<Product[]> {
+    return this.http.get<Product[]>(this.apiUrl);
+  }
+
+  get(id: number): Observable<Product> {
+    return this.http.get<Product>(`${this.apiUrl}/${id}`);
+  }
+
+  update(product: Product): Observable<Product> {
+    return this.http.patch<Product>(`${this.apiUrl}/edit-product/${product.id}`, product);
+  }
+
+  create(product: Product): Observable<Product> {
+    return this.http.post<Product>(this.apiUrl, product);
+  }
+
+  remove(product: Product): Observable<Product> {
```

Comment:
Egységes CRUD nevezéktant használjunk!

```suggestion
  delete(product: Product): Observable<Product> {
```

Reply:
Ugyanaz vonatkozik rá, mint az előzőre.